riptide-opentracing/README.md

```diff
@@ -0,0 +1,185 @@
+| `error`              | `ErrorSpanDecorator`           | `false`                           |
+| `error.kind` (log)   | `ErrorSpanDecorator`           | `SocketTimeoutException`          |
+| `error.object` (log) | `ErrorSpanDecorator`           | (exception instance)              |
+| `retry`              | `RetrySpanDecorator`           | `true`                            |
+| `retry_number` (log) | `RetrySpanDecorator`           | `3`                               |
+| `*`                  | `CallSiteTagSpanDecorator`     | `admin=true`                      |
+| `*`                  | `StaticTagSpanDecorator`       | `aws.region=eu-central-1`         |
+| `*`                  | `UriVariablesTagSpanDecorator` | user_id=me                        |
+
+### Notice
+
+**Be aware**: The `http.url` tag is disabled by default because the full request URI may contain
+sensitive, [*personal data*](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation).
+As an alternative we introduced the `http.path` tag which favors the URI template over the
+already expanded version. That has the additional benefit of a significant lower cardinality
+compared to what `http.url` would provide. 
+
+If you still want to enable it, you can do so by just registering the missing span decorator:
+
+```java
+new OpenTracingPlguin(tracer)
+    .withAdditionalSpanDecorators(new HttpUrlSpanDecorator())
+```
+
+### Span Decorators
+
+Span decorators are a simple, yet powerful tool to manipulate the span, i.e. they allow you to
+add tags, logs and baggage to spans. The default set of decorators can be extended by using 
+`OpenTracingPlugin#withAdditionalSpanDecorators(..)`:
+
+```java
+new OpenTracingPlguin(tracer)
+    .withAdditionalSpanDecorators(new StaticSpanDecorator(singletonMap(
+            "environment", "local"
+    )))
+```
+
+If the default span decorators are not desired you can replace them completely using
+`OpenTracingPlugin#withSpanDecorators(..)`:
+
+```java
+new OpenTracingPlguin(tracer)
+        .withSpanDecorators(
+            new ComponentSpanDecorator("MSIE"),
+            new SpanKindSpanDecorator(Tags.SPAN_KIND_CONSUMER),
+            new PeerSpanDecorator(),
+            new HttpMethodSpanDecorator(),
+            new HttpPathSpanDecorator(),
+            new HttpUrlSpanDecorator(),
+            new HttpStatusCodeSpanDecorator(),
+            new ErrorSpanDecorator(),
+            new CallSiteTagSpanDecorator())
+```
+
+## Usage
+
+Typically you won't need to do anything at the call-site regarding OpenTracing, i.e.
+you usages of Riptide should work exactly as before:
+
+```java
+http.get("/users/{id}", userId)
+    .dispatch(series(),
+        on(SUCCESSFUL).call(User.class, this::greet),
+        anySeries().call(problemHandling()))
+```
+
+### Operation Name
+
+By default the HTTP method will be used as the operation name, which might not fit your needs.
```

Comment:
Have you considered making the operation name more of a first-class citizen and an argument for the method `get()` (in the example)?

Reply:
That would break our modularization concept and would require OpenTracing to be part of the core rather than a separate add-on module.

Reply:
Understood. Sounds reasonable.
In a separate initiative (universal RPC), we're thinking about enforcing naming the operation. This already seemed very close to what we've come up with so far, that's why I was asking